src/init.cpp

```diff
@@ -384,7 +384,7 @@ void SetupServerArgs()
             "Warning: Reverting this setting requires re-downloading the entire blockchain. "
             "(default: 0 = disable pruning blocks, 1 = allow manual pruning via RPC, >%u = automatically prune block files to stay under the specified target size in MiB)", MIN_DISK_SPACE_FOR_BLOCK_FILES / 1024 / 1024), false, OptionsCategory::OPTIONS);
     gArgs.AddArg("-reindex", "Rebuild chain state and block index from the blk*.dat files on disk", false, OptionsCategory::OPTIONS);
-    gArgs.AddArg("-reindex-chainstate", "Rebuild chain state from the currently indexed blocks", false, OptionsCategory::OPTIONS);
+    gArgs.AddArg("-reindex-chainstate", "Rebuild chain state from the currently indexed blocks. When in pruning mode or if blocks on disk might be corrupted, use full -reindex instead.", false, OptionsCategory::OPTIONS);
```

Comment:
I believe it will automatically prompt you to use `-reindex` when needed while in pruning mode, so maybe no need to mention it here?

Then, you could move the note about corruption to reindex. E.g. "Rebuild chain state and block index from the blk*.dat files on disk *. Only needed when blocks on disk are corrupted.* "

Reply:
I just sabotaged a pruned node by deleting the most recent `blkxxxxx.dat` file and then started bitcoind: `Please restart with -reindex or -reindex-chainstate to recover.`

Restarting that pruned node with `-reindex-chainstate` will result in the same error message and won't tell the user they need to use `-reindex` instead. So I think it's better to mention pruned node in the help here.